docs/examples/expressions/expr_conditional_and_masking.ipynb

```diff
@@ -0,0 +1,386 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Conditional-like execution and masking\n",
+    "\n",
+    "This notebook shows how DALI arithmetic expressions can be utilized to achieve conditional-like application of augmentations and be used for some of the masking operations."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Conditional results\n",
+    "\n",
+    "We will create a Pipeline that will utilize DALI arithemtic expressions to apply augmentations conditionally its inputs. As DALI does not support executing parts of the graph in the Pipeline we will emulate the conditional operator in a similar fashion to [muxing](https://en.wikipedia.org/wiki/Multiplexer). Keep in mind that all possible inputs to our mux implementation will still be calculated by DALI.\n",
+    "\n",
+    "#### Imports\n",
+    "\n",
+    "Lets start with necessary imports."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from __future__ import division\n",
+    "from nvidia.dali.pipeline import Pipeline\n",
+    "import nvidia.dali.ops as ops\n",
+    "import nvidia.dali.types as types\n",
+    "from nvidia.dali.types import Constant"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Operators used explicitly\n",
+    "We don't need to explicitly list arithmetic operators that we want to use in the Pipeline constructor. They work as regular Python operators in the `define_graph` step. \n",
+    "\n",
+    "As for the rest of Operators, our pipeline will use FileReader to provide us with input images. We also need an ImageDecoder to decode the loaded images.\n",
+    "\n",
+    "We will use CoinFlip as source for random condition. We will cast the result to bool type, so it will play nicely with the type promotion rules.\n",
+    "\n",
+    "As an example augmentation we will apply the BrightnessContrast Operator. We choose quite extreme parameters, so it will clearly show in the output."
```

Comment:
```suggestion
    "As an example augmentation, we will apply the *BrightnessContrast* Operator. We choose quite extreme parameters, so it will clearly show in the output."
```

Reply:
Done